media-source-respec.html

```diff
@@ -599,6 +688,40 @@ <h3>Event Summary</h3>
         </table>
       </section>
 
+      <section id="mediasource-in-worker-communication-model">
+        <h3><dfn>Cross-context communication model</dfn></h3>
+
+        <p>When a {{Window}} {{HTMLMediaElement}} is attached to a {{DedicatedWorkerGlobalScope}} {{MediaSource}}, each context
+        has algorithms that depend on information from the other.</p>
+
+        <p class="note">{{HTMLMediaElement}} is exposed only to {{Window}} contexts, but {{MediaSource}} and related objects defined
+        in this specification are exposed in {{Window}} and {{DedicatedWorkerGlobalScope}} contexts. This lets
+        applications construct a {{MediaSource}} object in either of those types of context and attach it to an
+        {{HTMLMediaElement}} object in a {{Window}} context using a [=MediaSource object URL=] as described in the
+        [=attaching to a media element=] algorithm. A {{MediaSource}} object is not {{Transferable}}; it is only visible
+        in the context where it was created.</p>
+
+        The rest of this section describes a model for bounding information latency for attachments of a {{Window}}
+        media element to a {{DedicatedWorkerGlobalScope}} {{MediaSource}}. While the model describes communication using
+        message passing, implementations MAY choose to communicate in potentially faster ways, such as using shared
+        memory and locks. Attachments to a {{Window}} {{MediaSource}} synchronously have the information already without
+        communicating it across contexts.
+
+        <p>A {{MediaSource}} that is constructed in a {{DedicatedWorkerGlobalScope}} has an internal, unexposed
+        <dfn id="port-to-main">port to main</dfn> nullable variable that stores a {{MessagePort}} setup during
```

Comment:
In other specs, these internal, unexposed variable are called [internal slots](https://medium.com/jspoint/what-are-internal-slots-and-internal-methods-in-javascript-f2f0f6b38de) and rendered as `[[port to main]]`. From a ReSpec perspective, dfn would be `<dfn data-dfn-for="MediaSource">[[\port to main]]</dfn> and references would use `{{MediaSource/[[port to main]]}}`

Reply:
Yes I've seen the internal slot spec abstraction occurring recently in other specs. I'll update this feature draft for port-to-main/window and channel to be held in explicit internal slots. Separately, in a different PR, I'll update the existing object variables like "first initialization segment received flag" to be held in internal slots, too.

Reply:
98fe6fc should fix this.